bundles/org.openhab.ui/web/src/components/item/item-form.vue

```diff
@@ -70,6 +78,36 @@ export default {
           }
         }
       })
+    },
+    onNameInput (event) {
+      this.item.name = event.target.value
+      this.validateName(this.item.name)
+    },
+    validateName (name) {
+      let oldError = this.nameErrorMessage
+      if (!/^[A-Za-z0-9_]+$/.test(name)) {
+        this.nameErrorMessage = 'Required. Alphanumeric & underscores only'
+      } else if (this.allItems.some(item => item.name === name)) {
+        this.nameErrorMessage = 'An item with this name already exists'
+      } else {
+        this.nameErrorMessage = ''
+      }
+      if (oldError !== this.nameErrorMessage) this.$emit('valid', !this.nameErrorMessage)
+    }
+  },
+  created () {
+    if (this.enableName) {
+      if (!this.items) {
+        this.$oh.api.get('/rest/items').then((items) => {
```

Comment:
Is there a case where the items are not supplied by the parent component in the "items" prop and they have to be loaded by this component?
If the answer is no then maybe we can assume they will always be supplied, and remove this logic and the changes to the templates to include the preloader on lines 3-7?

Reply:
> Is there a case where the items are not supplied by the parent component in the "items" prop and they have to be loaded by this component?

There actually seem to be many: add-from-thing, item-edit, channel-list, ... They all don't load items to forward.

Reply:
So say you want to "add a thing as equipment" which has 200 channels, click on Select All to check all these channels causing the item forms to render, it's going to call `/rest/items` 200 times and keep separate versions of the whole list in the individual item form instances? This doesn't seem right.